### PR TITLE
GitHub Workflow

### PR DESCRIPTION
**Pull request by joshmoore**
Run tests automatically.

Possible TODOs:
 * [ ] use pre-created images to speed up processing

### CODE REVIEW

**Comment by joshmoore** [comment]
_Whew_ https://github.com/joshmoore/ome_zarr_test_suite/actions/runs/767280007 is green after filtering out omero-ms-zarr. (Which means omero-cli-zarr will likely not be runnable via GitHub Actions either, @dominikl)

cc: @jburel

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          auto-update-conda: true
+          channels: conda-forge,ome
+          environment-file: environment.yml
+          python-version: 3.9
+
+      - name: Install dependencies (temp)
+        shell: bash -l {0}
+        run: |
+          python -m pip install \
+          git+https://github.com/joshmoore/ome-zarr-py.git@nested
```
Comment (jburel) [inline .github/workflows/main.yml:29]: similar to benchmark repo, is the branch ready for review/merge into mainline

Comment (joshmoore) [reply .github/workflows/main.yml:29]: cc: @will-moore

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+dependencies:
+  - bioformats2raw
+  - python == 3.9.0
+  - wheel
+  - s3fs
+  - zarr >= 2.7.0
+  - pip
+  - pip:
+      - ome-zarr
+      - pytest
```
Comment (jburel) [inline environment.yml:14]: any reason not to install pytest via conda i.e. https://anaconda.org/anaconda/pytest
I am still thinking if we want to have a conda.lock at some point, mixing pip and conda implies that the deployment become a 2 steps process.

File: suites.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 ---
-- script: ome_zarr_info
-- script: ome_zarr_downloads
-- script: zarr_checksum
+- name: info
+  script: ome_zarr_info
+
+- name: download
```
Comment (jburel) [inline suites.yml:5]: ``downloads`` to match the name of the script

File: environment.yml (hunk 2 above)
Comment (joshmoore) [reply environment.yml:14]: Happy to move it up, but I wouldn't pin here since we _want_ it to fail. We could have one of each though.

**Comment by joshmoore** [comment]
`No runner matching the specified labels was found: self-hosted.` Hmmm...

**Comment by jburel** [comment]
Looks good. I think the TODO can be looked at afterwards. 
Better to merge so the action is in place for this repository

**Pull request merged**